framework/base/src/storage/mappers/queue_mapper.rs

```diff
@@ -62,34 +73,37 @@ impl QueueMapperInfo {
 ///
 /// The `QueueMapper` allows pushing and popping elements at either end
 /// in constant time.
-pub struct QueueMapper<SA, T>
+pub struct QueueMapper<SA, T, A = StorageSCAddress>
 where
     SA: StorageMapperApi,
-    T: TopEncode + TopDecode + 'static,
+    A: StorageAddress<SA>,
+    T: TopEncode + TopDecode + NestedEncode + NestedDecode + 'static,
```

Comment:
Why is nested encoding needed here?

Reply:
It is not, has been removed in the last commit